### ISSUE TITLE
Confusion about the comment in the sampling method

### ISSUE THREAD

**Issue by zhenzhang1998** [SysCV/sam-hq#37]
I am confused with the below setences... In `train/utils/loss_mask.py Line67`
```python
    # It is crucial to calculate uncertainty based on the sampled prediction value for the points.
    # Calculating uncertainties of the coarse predictions first and sampling them for points leads
    # to incorrect results.
    # To illustrate this: assume uncertainty_func(logits)=-abs(logits), a sampled point between
    # two coarse predictions with -1 and 1 logits has 0 logits, and therefore 0 uncertainty value.
    # However, if we calculate uncertainties for the coarse predictions first,
    # both will have -1 uncertainty, and the sampled point will get -1 uncertainty.
```

Why the two coarse predictions with -1 and 1 logits has 0 logits, and therefore 0 uncertainty value.  Is the bilinear upsample caused?

**Comment by lkeab** [comment]
Hi，this function comments explain the point sampling strategies when computing mask losses. It is "a sampled point between two coarse predictions with -1 and 1 logits has 0 logits", '0 logits' indicates most uncertain point thus it achieves maximal value for function -abs(logits).

**Comment by zhenzhang1998** [comment]
So, Why a sampled point between two coarse predictions with -1 and 1 logits has 0 logits ? Is the bilinear upsample caused?
